src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -1659,7 +1662,7 @@ private bool TryGenerateSimplifiedGo(RegexNode node)
             }
 
             // Declare some locals.
-            LocalBuilder runtextLocal = DeclareString();
+            LocalBuilder runtextSpanLocal = DeclareReadOnlySpanChar();
```

Comment:
@stephentoub I noticed that SimplifiedGo() is using separate locals as opposed to using the class fields (e.g. `_runtextLocal`) Is there a reason for that? If not I can fix that in a subsequent PR.

Reply:
Perf, I bet.

Reply:
The fields go away in #62318. I used locals for a few reasons. I kept having issues where I'd end up using a field initialized for one method (e.g. FindFirstChar) in another (e.g. Go); with fields there was no distinction as to which method they were for.  With nullable reference types, as fields they needed to be nullable, which meant lots of `!`s, whereas locals for the most part the compiler's flow analysis can see they've been initialized.  And since locals were needed for other things, it increased the consistency to have everything just be locals.